### PR TITLE
PL-163: Changing pdfs is not resetting properly

### PR DESCRIPTION
**Pull request by tonesija**
DONE:
Fixed bug when pages not rendering properly when changing pdfs 
Correct page when switching pdf
Knowledge graph and summary when switching pdf reset 
TODO:
Close reference view when switching pdfs

### CODE REVIEW

**Comment by tonesija** [comment]
I also refactored loadPdf to use async/await as an example of what we should be doing.

**Comment by NinaGnjidic** [comment]
All’s good, I just have a question/suggestion… Since we have onResetReader event, how about subscribing other components that are now being reset in reset method to that event and reseting them that way? How everything is now is okay, and ofc you can leave it as it is, I’m just saying this because that way reader is not resposnible for resetting other components but only itself 🤷‍♀️
That could also potentially solve the problem with reference view, but I can help you solve it because I already have some ideas how to…
As I said you can leave things as they are if you want to, this is just an idea and I’m only really asking your opinion on this…

File: src/components/PdfReaderComponent.js [hunk 1]
```diff
@@ -99,28 +99,25 @@ class PdfReaderComponent {
   };
 
   /**
-   * Load and render the first page of the given pdf.
+   * Loads the pdf document, configures 'child' components, delegates
+   * page setup and rendering of the first page.
    * @param {Uint8Array} pdf data, filename or url of a PDF document
    */
-  loadPdf = (pdf) => {
-    const self = this;
```
Comment (NinaGnjidic) [inline src/components/PdfReaderComponent.js:106]: I didnt even think of this benefit of using async await, but we even lost “self” which is 👌

**Comment by tonesija** [comment]
> All’s good, I just have a question/suggestion… Since we have onResetReader event, how about subscribing other components that are now being reset in reset method to that event and reseting them that way? How everything is now is okay, and ofc you can leave it as it is, I’m just saying this because that way reader is not resposnible for resetting other components but only itself 🤷‍♀️ That could also potentially solve the problem with reference view, but I can help you solve it because I already have some ideas how to… As I said you can leave things as they are if you want to, this is just an idea and I’m only really asking your opinion on this…

We can solve that in the refactoring tasks imo. I 100% agree with you with what you said, it will decouple components better.

**Review by Giamburrasca — COMMENTED** [review]
Except for the refactor as Nina said, it works, for me +1

**Comment by NinaGnjidic** [comment]
Tested locally and it works how it's supposed to, code is okay and it'll be refactored in PL-155, so +1

**Comment by paolocorsa** [comment]
> Tested locally and it works how it's supposed to, code is okay and it'll be refactored in PL-155, so +1

I have nothing to add, +1

**Pull request merged**